client.go

```diff
@@ -30,104 +31,36 @@ type Client struct {
 	Net            string            // if "tcp" or "tcp-tls" (DNS over TLS) a TCP query will be initiated, otherwise an UDP one (default is "" for UDP)
 	UDPSize        uint16            // minimum receive buffer for UDP messages
 	TLSConfig      *tls.Config       // TLS connection configuration
-	Timeout        time.Duration     // a cumulative timeout for dial, write and read, defaults to 0 (disabled) - overrides DialTimeout, ReadTimeout and WriteTimeout when non-zero
-	DialTimeout    time.Duration     // net.DialTimeout, defaults to 2 seconds - overridden by Timeout when that value is non-zero
+	Timeout        time.Duration     // a cumulative timeout for dial, write and read, defaults to 0 (disabled) - overrides DialTimeout, ReadTimeout, WriteTimeout when non-zero. Can be overridden with net.Dialer.Timeout (see Client.ExchangeWithDialer and Client.DialWithDialer) or context.Context.Deadline (see the deprecated ExchangeContext)
```

Comment:
Wouldn't it be better to just put a Dialer in here? Why not?

Reply:
The only reason was to avoid breaking code that uses this (public) interface. Anyone creating a `Client` with a Timeout would see their code failing the build.
What to do here depends on how inclined you are to push a breaking change

Reply:
I don't meaning removing anything, just *adding* a net.Dailer in here...

Reply:
...I have no idea why I didn't consider this since the beginning. OK, will add a Dialer here. This means that I can remove `Client.DialWithDialer` and `Client.ExchangeWithDialer` and just leave `Client.Dial` and `Client.Exchange` that will implicitly use the `Client.Dialer` if it's not nil. Does this sound right?

Reply:
done, diff incoming